### DISCUSSION TITLE
Implementing BrighterScript within a Component Library?

### DISCUSSION THREAD

**Discussion by willmatos** [rokucommunity/brighterscript#440] [Q&A] [answered]
Hi guys! Thanks for putting together brighterscript. We would love to use it in a pretty big app that we have but the problem is that all groups that contributed to the app have not converted to Brighertscript. I was wondering if anyone in this group had tried to implement BrighterScript within just a component library?

**Comment by markwpearce** [comment, marked as answer]
There's no reason why you couldn't include a component library that was developed/written in Brighterscript. Brighterscript just transpiles down to regular old Brightscript, so as long as the compiled version is what is being deployed, I don't think there would be a problem.

Also note that Brighterscript and Brightscript can live beside each other in the same codebase. The Brighterscript stuff gets transpiled to brightscript, anyway, and the VScode plugin handles both.

**Reply by willmatos** [reply]
Thanks Mark! We did a test run with a basic Component Library that we made from scratch just as POC and it worked out. There just must be something within our component library that isn't playing well. Not compiling errors but also not playing how it should with the repo thats consuming it.

**Comment by elsassph** [comment]
I might suggest to check my Markdown component which is a Brighterscript project and which we use in a vanilla project: https://github.com/elsassph/roku-markdown
